test/t8n/t8n.cpp

```diff
@@ -0,0 +1,172 @@
+
+            evmc::VM vm{evmc_create_evmone(), {{"O", "0"}}};
+
+            std::vector<state::Log> txs_logs;
+
+            if (j_txs.is_array())
+            {
+                j_result["receipts"] = json::json::array();
+                j_result["rejected"] = json::json::array();
+                auto idx = 0;
+                for (auto& j_tx : j_txs)
+                {
+                    state::Transaction tx = test::from_json<state::Transaction>(j_tx);
+                    auto const [tx_logs, gas_used] = state::transition(state, block, tx, rev, vm);
+                    if (tx_logs.has_value())
+                    {
+                        txs_logs.insert(txs_logs.begin(), (*tx_logs).begin(), (*tx_logs).end());
+                        auto& j_receipt = j_result["receipts"][j_result["receipts"].size()];
+                        j_receipt["transactionHash"] = j_tx["hash"];
+                        j_receipt["gasUsed"] = tu::shortest_hex(uint64_t(gas_used));
+                        j_receipt["cumulativeGasUsed"] = j_receipt["gasUsed"];
+
+                        j_receipt["blockHash"] = NULL_HEXSTRING_32;
+                        j_receipt["contractAddress"] = NULL_HEXSTRING_20;
+                        j_receipt["logsBloom"] = NULL_HEXSTRING_256;
+                        j_receipt["logs"] = json::json::array();  // FIXME: Add to_json<state:Log>
+                        j_receipt["root"] = "";
+                        j_receipt["status"] = "0x1";
+                        j_receipt["transactionIndex"] = "0x0";
+                    }
+                    else
+                    {
+                        json::json j_rejected_tx;
+                        j_rejected_tx["hash"] = j_tx["hash"];
+                        j_rejected_tx["index"] = idx;
+                        j_rejected_tx["error"] = "TODO: Implement returning errors";
+                        j_result["rejected"].push_back(j_rejected_tx);
+                    }
+
+                    idx++;
+                }
+            }
+
+            j_result["logsHash"] = tu::hex(keccak256(rlp::encode(txs_logs)));
+            j_result["stateRoot"] = tu::hex(state::mpt_hash(state.get_accounts()));
+        }
+
+        j_result["logsBloom"] = NULL_HEXSTRING_256;
+        j_result["receiptsRoot"] = NULL_HEXSTRING_32;
+        j_result["txRoot"] = NULL_HEXSTRING_32;
+
+        std::ofstream{output_dir / output_result_file} << j_result;
+
+        // Print out current state to outAlloc file
+        json::json j_alloc;
+        for (const auto& [addr, acc] : state.get_accounts())
+        {
+            j_alloc[tu::hex(addr)]["nonce"] = test::utils::shortest_hex(acc.nonce);
+            for (const auto& [key, val] : acc.storage)
+                if (!is_zero(val.current))
+                    j_alloc[tu::hex(addr)]["storage"][tu::hex(key)] = tu::hex(val.current);
+
+            j_alloc[tu::hex(addr)]["code"] = tu::hex({acc.code.data(), acc.code.size()});
```

Comment:
This has no `0x` prefix?

Reply:
One more idea: we could have an overloaded helper `hex` defined in this file at the top for each type we care of, so that the body here can just use `hex()` solely. This way we can inspect what types are used and how they get translated.

Reply:
But if these helper needs to use hex impl from different libs like evmc it will be a mess with specifying which one we want to use.

Reply:
I meant something like this:
```cpp
namespace {
inline bytes hex(string_view s) {
    return evmc::hex(s);
}
inline bytes hex(intx::u256 v) {
    return intx::hex(v);
}
}
```

Reply:
Sure I understand but we will have to remove a couple of `using namespace evmc` etc.